### PR TITLE
[feature-array](array-type) Add array function array_with_constant

### PR DESCRIPTION
**Pull request by eldenmoon**
return array of constants with length num.
```
mysql> select array_with_constant(4, 1223);
+------------------------------+
| array_with_constant(4, 1223) |
+------------------------------+
| [1223, 1223, 1223, 1223]     |
+------------------------------+
1 row in set (0.01 sec)
```

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.86 seconds
 load time: 498 seconds
 storage size: 17179226064 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221109105825_clickbench_pr_42693.html

File: regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy [hunk 1]
```diff
@@ -70,4 +70,8 @@ suite("test_array_functions") {
     qt_select "SELECT k1, array_popback(k5) from ${tableName} ORDER BY k1"
     qt_select "SELECT k1, array_popback(k6) from ${tableName} ORDER BY k1"
     qt_select "SELECT k1, array_popback(k7) from ${tableName} ORDER BY k1"
+    qt_select "SELECT k1, array_with_constant(3, k1) from ${tableName} ORDER BY k1"
+    qt_select "SELECT k1, array_with_constant(10, null) from ${tableName} ORDER BY k1"
```
Comment (xy720) [inline regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy:74]: Why the result is null here?

File: be/src/vec/functions/array/function_array_with_constant.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            }
+            offset += array_size;
+            offsets.push_back(offset);
+            array_sizes.push_back(array_size);
+        }
+        auto clone = value->clone_empty();
+        clone->reserve(input_rows_count);
+        value->replicate(array_sizes.data(), offset, *clone->assume_mutable().get());
+        if (value->is_nullable()) {
+            auto nested_column = ColumnNullable::create(clone->assume_mutable(),
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply be/src/vec/functions/array/function_array_with_constant.cpp]: fixed, use [ALWAYS_NOT_NULLABLE] to make the return type more precise

File: be/src/vec/functions/array/function_array_with_constant.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    bool is_variadic() const override { return false; }
+
+    size_t get_number_of_arguments() const override { return 2; }
+
+    // need handle null cases
+    bool use_default_implementation_for_nulls() const override { return false; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        if (arguments[1]->is_nullable()) {
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply be/src/vec/functions/array/function_array_with_constant.cpp]: fixed, use [ALWAYS_NOT_NULLABLE] to make the return type more precise

File: be/src/vec/functions/array/function_array_with_constant.cpp [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                                            get_name());
+            }
+            offset += array_size;
+            offsets.push_back(offset);
+            array_sizes.push_back(array_size);
+        }
+        auto clone = value->clone_empty();
+        clone->reserve(input_rows_count);
+        value->replicate(array_sizes.data(), offset, *clone->assume_mutable().get());
+        if (value->is_nullable()) {
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply be/src/vec/functions/array/function_array_with_constant.cpp]: fixed, use [ALWAYS_NOT_NULLABLE] to make the return type more precise

File: regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy (hunk 1 above)
Comment (eldenmoon) [reply regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy:74]: fixed, use [ALWAYS_NOT_NULLABLE] to make the return type more precise

File: docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_with_constant.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+1 row in set (0.04 sec)
+
+mysql> select array_with_constant(3, 12345);
++-------------------------------+
+| array_with_constant(3, 12345) |
++-------------------------------+
+| [12345, 12345, 12345]         |
++-------------------------------+
+1 row in set (0.01 sec)
+
```
Comment (xy720) [inline docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_with_constant.md:60]: Please add explain and example for case array_with_constant(3, null) in document.

Comment (eldenmoon) [reply docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_with_constant.md:60]: added

**Review by xy720 — APPROVED** [review]
LGTM

**Pull request merged**